test/connectWithShell.spec.tsx

```diff
@@ -636,6 +637,116 @@ describe('connectWithShell-useCases', () => {
         expect(hasPendingSubscribers()).toBe(false)
     })
 
+    it('should not notify subscribers when deferring notifications', async () => {
+        const { host, shell, renderInShellContext } = createMocks(entryPointWithState, [entryPointSecondStateWithAPI])
+        const ConnectedComp = connectWithShell(mapStateToProps, undefined, shell, { allowOutOfEntryPoint: true })(PureComp)
+
+        const { testKit } = renderInShellContext(<ConnectedComp />)
+        if (!testKit) {
+            throw new Error('Connected component failed to render')
+        }
+
+        expect(testKit.root.findByProps({ className: 'ONE' }).children[0]).toBe('init1')
+        expect(mapStateToPropsSpy).toHaveBeenCalledTimes(1)
+        expect(renderSpy).toHaveBeenCalledTimes(1)
+
+        let valueOneWhileDeferring
+
+        await act(async () => {
+            await host.getStore().deferSubscriberNotifications(() => {
+                dispatchAndFlush({ type: 'SET_FIRST_STATE', value: 'update1' }, host)
+                valueOneWhileDeferring = testKit.root.findByProps({ className: 'ONE' }).children[0]
+            })
+        })
+
+        expect(valueOneWhileDeferring).toBe('init1')
+        expect(testKit.root.findByProps({ className: 'ONE' }).children[0]).toBe('update1')
+        expect(mapStateToPropsSpy).toHaveBeenCalledTimes(2)
+        expect(renderSpy).toHaveBeenCalledTimes(2)
+    })
+
+    it('should notify after action failed while deferring notifications', async () => {
+        const { host, shell, renderInShellContext } = createMocks(entryPointWithState, [entryPointSecondStateWithAPI])
+        const ConnectedComp = connectWithShell(mapStateToProps, undefined, shell, { allowOutOfEntryPoint: true })(PureComp)
+
+        const { testKit } = renderInShellContext(<ConnectedComp />)
+        if (!testKit) {
+            throw new Error('Connected component failed to render')
+        }
+
+        expect(testKit.root.findByProps({ className: 'ONE' }).children[0]).toBe('init1')
```

Comment:
Maybe you can extract `testKit.root.findByProps({ className: 'ONE' }).children[0]` to a function 🙃

Reply:
Absolutely :)